.github/workflows/lint-bc.yml

```diff
@@ -1,15 +1,25 @@
 name: BC Lint
 
 on:
+  # Copied from check-labels.yml to get around needing approval for first time contributors
+  # We need pull_request_target to be able to post comments on PRs from forks.
+  # See https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks
+  # Only allow pull_request_target when merging to main, not some historical branch.
+  #
+  # Make sure to don't introduce explicit checking out and installing/running
+  # untrusted user code into this workflow!
+  pull_request_target:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+    branches: [main]
+    paths-ignore: [.github]
+
+  # To allow testing PRs that change workflows.
+  # May be triggered together with pull_request_target, it's OK.
   pull_request:
-    types:
-      - opened
-      - synchronize
-      - reopened
-      - labeled
-      - unlabeled
-    branches-ignore:
-      - nightly
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+    paths: [.github]
```

Comment:
May be set the `pull_requests` paths explicitly to `.github/workflows/lint-bc.yaml`, but this is probably fine too